### PR TITLE
AEA-267 fix programmatic guide, add programmatic mode flag to aea/agent

### LINKED ISSUE #619: Hacking an agent guide no longer working
**Is your feature request related to a problem? Please describe.**
The guide is no longer working.

**Describe the solution you'd like**
Make changes so it works again.

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR fixes the guide to build an agent programmatically. It also adds a flag to an AEA/Agent which lets the developer skip the loading from resource directory stage.

## Fixes

#619 

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../master/CONTRIBUTING.rst) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: aea/aea.py [hunk 1]
```diff
@@ -48,6 +48,7 @@ def __init__(self, name: str,
                  loop: Optional[AbstractEventLoop] = None,
                  timeout: float = 0.0,
                  debug: bool = False,
+                 programmatic: bool = True,
```
Comment (panasevychol) [inline aea/aea.py]: I prefer more self explaining naming (`is_programmatic` in this case).

File: tests/test_aea.py [hunk 2]
```diff
@@ -72,13 +73,15 @@ def test_act():
         ledger_apis = LedgerApis({}, 'default')
         address = wallet.addresses['default']
         connections = [OEFLocalConnection(address, node)]
+        resources = Resources(str(Path(CUR_PATH, "data", "dummy_aea")))
```
Comment (panasevychol) [inline tests/test_aea.py]: Could be replaced with `os.path.join(...)`. Returns str so doesn't require an extra action - to lead it to a str.

Comment (panasevychol) [reply tests/test_aea.py]: If that's a common action (I've seen this before) I would request to upgrade `Path` with `to_str()` method.

File: tests/test_skills/test_base.py [hunk 3]
```diff
@@ -61,7 +61,7 @@ def setup_class(cls):
         cls.wallet = Wallet({'default': private_key_pem_path, FETCHAI: private_key_txt_path})
         cls.ledger_apis = LedgerApis({FETCHAI: ["alpha.fetch-ai.com", 80]}, FETCHAI)
         cls.connections = [DummyConnection()]
-        cls.my_aea = AEA("Agent0", cls.connections, cls.wallet, cls.ledger_apis, resources=Resources(str(Path(CUR_PATH, "data", "dummy_aea"))))
+        cls.my_aea = AEA("Agent0", cls.connections, cls.wallet, cls.ledger_apis, resources=Resources(str(Path(CUR_PATH, "data", "dummy_aea"))), programmatic=False)
```
Comment (panasevychol) [inline tests/test_skills/test_base.py]: It's quite a common case.

File: tests/test_aea.py (hunk 2 above)
Comment (DavidMinarsch) [reply tests/test_aea.py]: Thanks, I agree. But this is applied throughout the entire project. So I won't change it in this PR.

File: aea/aea.py (hunk 1 above)
Comment (DavidMinarsch) [reply aea/aea.py]: Good point. I will open an issue as we should change it systematically for all variables of boolean type.

**Pull request merged**